python/cuml/svm/linear_svc.py

```diff
@@ -0,0 +1,187 @@
+        while ``'ovr'`` selects `OneVsRestClassifier
+        <https://scikit-learn.org/stable/modules/generated/sklearn.multiclass.OneVsRestClassifier.html>`_
+    output_type : {{'input', 'cudf', 'cupy', 'numpy', 'numba'}}, default=None
+        Variable to control output type of the results and attributes of
+        the estimator. If None, it'll inherit the output type set at the
+        module level, `cuml.global_settings.output_type`.
+        See :ref:`output-data-type-configuration` for more info.
+
+    Attributes
+    ----------
+    intercept_ : float, shape (n_classes,)
+        The constant in the decision function
+    coef_ : float, shape (n_classes, n_cols)
+        The vectors defining the hyperplanes that separate the classes.
+    classes_: float, shape (n_classes_,)
+        Array of class labels.
+    probScale_: float, shape (n_classes_, 2)
+        Probability calibration constants (for the probabolistic output).
+    n_classes_ : int
+        Number of classes
+
+    Notes
+    -----
+    The model uses the quasi-newton (QN) solver to find the solution in the
+    primal space. Thus, in contrast to generic :class:`SVC<cuml.svm.SVC>`
+    model, it does not compute the support coefficients/vectors.
+
+    Check the solver's documentation for more details
+    :class:`Quasi-Newton (L-BFGS/OWL-QN)<cuml.QN>`.
+
+    For additional docs, see `scikitlearn's LinearSVC
+    <https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html>`_.
+    '''
+
+    REGISTERED_LOSSES = set([
+        'hinge',
+        'squared_hinge'])
+
+    def __init__(self, **kwargs):
+        # NB: the keyword arguments are filtered in python/cuml/svm/linear.pyx
+        #     the default parameter values are reexported from
+        #                                      cpp/include/cuml/svm/linear.hpp
+        # set classification-specific defaults
+        if 'loss' not in kwargs:
+            kwargs['loss'] = 'squared_hinge'
+        if 'multiclass_strategy' not in kwargs:
+            kwargs['multiclass_strategy'] = 'ovr'
+
+        super().__init__(**kwargs)
+
+    @property
+    def loss(self):
+        return self.__loss
+
+    @loss.setter
+    def loss(self, loss: str):
+        if loss not in self.REGISTERED_LOSSES:
+            raise ValueError(
+                f"Classification loss type "
+                f"must be one of {self.REGISTERED_LOSSES}, "
+                f"but given '{loss}'.")
+        self.__loss = loss
+
+    def get_param_names(self):
+        return [
+            "handle",
+            "verbose",
+            "output_type",
+            'penalty',
+            'loss',
+            'fit_intercept',
+            'penalized_intercept',
+            'probability',
+            'max_iter',
+            'linesearch_max_iter',
+            'lbfgs_memory',
+            'C',
+            'grad_tol',
+            'change_tol',
+            'multiclass_strategy',
```

Comment:
The estimator [guide suggest](https://github.com/rapidsai/cuml/blob/branch-21.12/wiki/python/ESTIMATOR_GUIDE.md#implementing-get_param_names):
> Be sure to include super().get_param_names() in the returned list to properly set the super() attributes.

That would cover the case if the Base class gets another parameter added.

Reply:
Then the tests would fail anyway. I believe it's easier to read the source in this case if I list all of them explicitly.
Besides, `handle` and `verbose` are also parameters of the C++ `LinearSVM` model; if for some reason the `Base` class gets rid of one of those, I don't want them disappear from this list. To avoid that, I would need to have the union operation here,  which will look even messier.